### PR TITLE
add docs for supabase CLI completion

### PR DESCRIPTION
**Pull request by lyqht**
## What kind of change does this PR introduce?

Docs update

## Additional context

I have added [a Fig autocomplete spec for Supabase CLI](https://github.com/withfig/autocomplete/pull/1557), and initially thought to add it to supabase CLI's README, but since its not automated to keep to date to the supabase CLI's updates yet, @sweatybridge suggested to add it here in the docs.

### CODE REVIEW

File: spec/cli_v1_commands.yaml [hunk 1]
```diff
@@ -567,3 +567,17 @@ commands:
       ```sh
         -h, --help   help for create
       ```
+  - id: supabase-completion
```
Comment (kiwicopple) [inline spec/cli_v1_commands.yaml:570]: Hey @lyqht - I think this file is generated by the CLI repo, so it's possible that this one will get over-written unfortunately

Should we remove this file? (or add the change to the CLI repo?

Comment (lyqht) [reply spec/cli_v1_commands.yaml:570]: oh i see! can you point me to how Supabase CLI generates this? if it is generated how come the supabase-completion is left out?

File: apps/reference/_cli/intro.mdx [hunk 2]
```diff
@@ -16,6 +16,9 @@ The CLI is still under development, but it contains all the functionality for wo
 - CI/CD for releasing to production: [`supabase db push`](https://supabase.com/docs/reference/cli/usage#supabase-db-push)
 - Manage your Supabase projects: [`supabase projects`](https://supabase.com/docs/reference/cli/usage#supabase-projects)
 - Generate types directly from your database schema: [`supabase gen types`](https://supabase.com/docs/reference/cli/usage#supabase-gen)
+    - A [community-supported GitHub Action to generate Supabase DB types](https://github.com/lyqht/generate-supabase-db-types-github-action) also uses this command
+- Shell Autocomplete: [`supabase completion`](https://supabase.com/docs/reference/cli/usage#supabase-completion)
+    - A community Fig autocomplete spec for supabase is also available
```
[the remark this answers is not in the document]
Comment (lyqht) [reply apps/reference/_cli/intro.mdx]: added a link!

**Review by sweatybridge — APPROVED** [review]
LGTM with a few stylistic changes.

File: apps/reference/_cli/intro.mdx [hunk 3]
```diff
@@ -16,6 +16,9 @@ The CLI is still under development, but it contains all the functionality for wo
 - CI/CD for releasing to production: [`supabase db push`](https://supabase.com/docs/reference/cli/usage#supabase-db-push)
 - Manage your Supabase projects: [`supabase projects`](https://supabase.com/docs/reference/cli/usage#supabase-projects)
 - Generate types directly from your database schema: [`supabase gen types`](https://supabase.com/docs/reference/cli/usage#supabase-gen)
+    - A [community-supported GitHub Action to generate TypeScript types](https://github.com/lyqht/generate-supabase-db-types-github-action) using this command
+- Shell autocomplete: [`supabase completion`](https://supabase.com/docs/reference/cli/usage#supabase-completion)
+    - A [community-supported Fig autocomplete spec for Supabase](https://fig.io/manual/supabase) is also available
```
Comment (sweatybridge) [inline apps/reference/_cli/intro.mdx:21]:
```suggestion
    - A community-supported [GitHub Action to generate TypeScript types](https://github.com/lyqht/generate-supabase-db-types-github-action)
- Shell autocomplete: [`supabase completion`](https://supabase.com/docs/reference/cli/usage#supabase-completion)
    - A community-supported [Fig autocomplete spec for macOS terminal](https://fig.io/manual/supabase)
```

**Pull request merged**